Description: Remove websvg from TYPO3 core
 Websvg provides a swf file for emulating SVG on many browsers. This swf file
 can not be build with tools in main, therefore it is removed during build.
 This patch removes the embedding code for this swf from the TYPO3 core.
Author: Christian Welzel <gawain@camlann.de>
Last-Update: 2012-10-23

--- typo3.orig/typo3/sysext/core/Classes/Page/PageRenderer.php
+++ typo3/typo3/sysext/core/Classes/Page/PageRenderer.php
@@ -298,11 +298,6 @@
 	protected $extJsPath = 'contrib/extjs/';
 
 	/**
-	 * @var string
-	 */
-	protected $svgPath = 'contrib/websvg/';
-
-	/**
 	 * The local directory where one can find jQuery versions and plugins
 	 *
 	 * @var string
@@ -462,18 +457,6 @@
 	protected $endingSlash = '';
 
 	/**
-	 * SVG library
-	 *
-	 * @var boolean
-	 */
-	protected $addSvg = FALSE;
-
-	/**
-	 * @var boolean
-	 */
-	protected $enableSvgDebug = FALSE;
-
-	/**
 	 * Used by BE modules
 	 *
 	 * @var null|string
@@ -729,16 +712,6 @@
 		$this->extJsPath = $path;
 	}
 
-	/**
-	 * Sets Path for SVG library (websvg)
-	 *
-	 * @param string $path
-	 * @return void
-	 */
-	public function setSvgPath($path) {
-		$this->svgPath = $path;
-	}
-
 	/*****************************************************/
 	/*                                                   */
 	/*  Public Enablers / Disablers                      */
@@ -885,7 +858,6 @@
 		$this->enableExtCoreDebug = TRUE;
 		$this->enableExtJsDebug = TRUE;
 		$this->enableJqueryDebug = TRUE;
-		$this->enableSvgDebug = TRUE;
 	}
 
 	/*****************************************************/
@@ -1102,15 +1074,6 @@
 	}
 
 	/**
-	 * Gets Path for SVG library (relative to typo3 directory)
-	 *
-	 * @return string
-	 */
-	public function getSvgPath() {
-		return $this->svgPath;
-	}
-
-	/**
 	 * Gets the inline language labels.
 	 *
 	 * @return array The inline language labels
@@ -1705,33 +1668,6 @@
 	}
 
 	/**
-	 * Call function if you need the SVG library
-	 *
-	 * @return void
-	 */
-	public function loadSvg() {
-		$this->addSvg = TRUE;
-	}
-
-	/**
-	 * Call this function to load debug version of ExtJS. Use this for development only
-	 *
-	 * @return void
-	 */
-	public function enableSvgDebug() {
-		$this->enableSvgDebug = TRUE;
-	}
-
-	/**
-	 * Call this function to force flash usage with SVG library
-	 *
-	 * @return void
-	 */
-	public function svgForceFlash() {
-		$this->addMetaTag('<meta name="svg.render.forceflash" content="true" />');
-	}
-
-	/**
 	 * Call this function to load debug version of ExtJS. Use this for development only
 	 *
 	 * @return void
@@ -2074,7 +2010,7 @@
 
 	/**
 	 * Helper function for render the main JavaScript libraries,
-	 * currently: RequireJS, jQuery, PrototypeJS, Scriptaculous, SVG, ExtJs
+	 * currently: RequireJS, jQuery, PrototypeJS, Scriptaculous, ExtJs
 	 *
 	 * @return string Content with JavaScript libraries
 	 */
@@ -2089,9 +2025,6 @@
 			$out .= '<script src="' . $this->processJsFile(($this->backPath . $this->requireJsPath . 'require.js')) . '" type="text/javascript"></script>' . LF;
 		}
 
-		if ($this->addSvg) {
-			$out .= '<script src="' . $this->processJsFile(($this->backPath . $this->svgPath . 'svg.js')) . '" data-path="' . $this->backPath . $this->svgPath . '"' . ($this->enableSvgDebug ? ' data-debug="true"' : '') . '></script>' . LF;
-		}
 		// Include jQuery Core for each namespace, depending on the version and source
 		if (!empty($this->jQueryVersions)) {
 			foreach ($this->jQueryVersions as $namespace => $jQueryVersion) {
--- typo3.orig/typo3/sysext/frontend/Classes/ContentObject/ContentObjectRenderer.php
+++ typo3/typo3/sysext/frontend/Classes/ContentObject/ContentObjectRenderer.php
@@ -805,7 +805,6 @@
 			'SWFOBJECT' => 'ShockwaveFlashObject',
 			'FLOWPLAYER' => 'FlowPlayer',
 			'QTOBJECT' => 'QuicktimeObject',
-			'SVG' => 'ScalableVectorGraphics',
 			'EDITPANEL' => 'EditPanel',
 		);
 		$name = $classMapping[$name];
@@ -1204,16 +1203,6 @@
 		return $this->getContentObject('QTOBJECT')->render($conf);
 	}
 
-	/**
-	 * Rendering the cObject, SVG
-	 *
-	 * @param array $conf Array of TypoScript properties
-	 * @return string Output
-	 */
-	public function SVG($conf) {
-		return $this->getContentObject('SVG')->render($conf);
-	}
-
 	/************************************
 	 *
 	 * Various helper functions for content objects:
--- typo3.orig/typo3/sysext/frontend/Classes/Page/PageGenerator.php
+++ typo3/typo3/sysext/frontend/Classes/Page/PageGenerator.php
@@ -530,15 +530,6 @@
 		}
 		// Javascript Libraries
 		if (is_array($GLOBALS['TSFE']->pSetup['javascriptLibs.'])) {
-			if ($GLOBALS['TSFE']->pSetup['javascriptLibs.']['SVG']) {
-				$pageRenderer->loadSvg();
-				if ($GLOBALS['TSFE']->pSetup['javascriptLibs.']['SVG.']['debug']) {
-					$pageRenderer->enableSvgDebug();
-				}
-				if ($GLOBALS['TSFE']->pSetup['javascriptLibs.']['SVG.']['forceFlash']) {
-					$pageRenderer->svgForceFlash();
-				}
-			}
 			if ($GLOBALS['TSFE']->pSetup['javascriptLibs.']['Prototype']) {
 				$pageRenderer->loadPrototype();
 			}
